### PR TITLE
Fixing removal of integrator charms not removing the config options.

### PR DESCRIPTION
**Pull request by hyperbolic2346**
Need to reset the values of the variables for the kubelet and then restart.

Fixes: https://bugs.launchpad.net/charm-kubernetes-master/+bug/1822899

### CODE REVIEW

File: reactive/kubernetes_worker.py [hunk 1]
```diff
@@ -678,6 +678,10 @@ def configure_kubelet(dns, ingress_ip):
         kubelet_opts['cloud-provider'] = 'azure'
         kubelet_opts['cloud-config'] = str(kubelet_cloud_config_path)
         kubelet_opts['provider-id'] = azure.vm_id
+    else:
+        kubelet_opts['cloud-provider'] = ''
+        kubelet_opts['cloud-config'] = ''
+        kubelet_opts['provider-id'] = ''
```
Comment (Cynerva) [inline reactive/kubernetes_worker.py]: This shouldn't be necessary. When these args are not explicitly set, they will not be in kubelet_opts, so they will get automatically removed when [configure_kubernetes_service](https://github.com/charmed-kubernetes/layer-kubernetes-common/blob/cda05cfceb864f7488483e791470473db85cca36/lib/charms/layer/kubernetes_common.py#L369-L371) is called (if they were previously set).

File: reactive/kubernetes_worker.py [hunk 2]
```diff
@@ -1175,6 +1175,7 @@ def clear_cloud_flags():
     remove_state('kubernetes-worker.cloud.request-sent')
     remove_state('kubernetes-worker.cloud.blocked')
     remove_state('kubernetes-worker.cloud.ready')
+    set_state('kubernetes-worker.restart-needed')  # force restart
```
Comment (Cynerva) [inline reactive/kubernetes_worker.py]: The conditions for this handler:

```
@when_none('endpoint.aws.joined',
           'endpoint.gcp.joined',
           'endpoint.openstack.joined',
           'endpoint.vsphere.joined',
           'endpoint.azure.joined')
```

When kubernetes-worker isn't related to any cloud providers, this condition will always be true. As a consequence, every hook will set `kubernetes-worker.restart-needed`.

Comment (hyperbolic2346) [reply reactive/kubernetes_worker.py]: Yeah, I caught this yesterday and fixed it on the master, but forgot the worker. Fixing now.

**Comment by Cynerva** [comment]
LGTM, thanks :+1:

**Pull request merged**